frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segments_view.tsx

```diff
@@ -1080,60 +1084,60 @@ class SegmentsView extends React.Component<Props, State> {
     });
   };
 
-  handleChangeMeshVisibility = (layerName: string, groupId: number, isVisible: boolean) => {
-    const { visibleSegmentationLayer } = this.props;
-    if (visibleSegmentationLayer == null) return;
-    const segmentGroup = this.getSegmentsOfGroup(groupId);
-    if (segmentGroup == null) return;
-
-    segmentGroup.forEach((segment) => {
+  handleChangeMeshVisibilityInGroup = (layerName: string, groupId: number, isVisible: boolean) => {
+    this.handlePerSegment(groupId, (segment) => {
       if (Store.getState().localSegmentationData[layerName].isosurfaces[segment.id] != null) {
         Store.dispatch(updateIsosurfaceVisibilityAction(layerName, segment.id, isVisible));
       }
     });
 
-    const copyOfState = this.state.areSegmentsInGroupVisible;
-    copyOfState[groupId] = isVisible;
+    const newSegmentsInGroupVisible = {
+      ...this.state.areSegmentsInGroupVisible,
+      [groupId]: isVisible,
+    };
     this.setState({
-      areSegmentsInGroupVisible: copyOfState,
+      areSegmentsInGroupVisible: newSegmentsInGroupVisible,
     });
```

Comment:
are you sure that this is necessary? Since you are computing areSegmentsInGroupVisible in getDerivedStateFromProps, I would assume that the state is automatically updated as soon as the store changes (which is done when you update the visibility of an isosurface). I'd think, that getDerivedStateFromProps is the only place where  areSegmentsInGroupVisible needs to be set. Could you test this? If this is the case, the setState code here and in the other handle* functions can be removed.